autoload/go/guru.vim

```diff
@@ -198,23 +197,26 @@ function! s:async_guru(args) abort
   endfunction
 
   let start_options = {
-        \ 'close_cb': function("s:close_cb"),
+        \ 'close_cb': funcref("s:close_cb"),
+        \ 'in_mode': 'raw',
+        \ 'out_mode': 'nl',
+        \ 'err_mode': 'nl',
         \ }
 
-  if &modified
-    let l:tmpname = tempname()
-    call writefile(split(result.stdin_content, "\n"), l:tmpname, "b")
-    let l:start_options.in_io = "file"
-    let l:start_options.in_name = l:tmpname
-  endif
-
```

Comment:
For a few reasons: parity with how `guru_sync` works, symmetry with the command that `guru_cmd` constructs, and performance (it's faster to write to the current buffer's contents to guru's `stdin` than to write a temp file).

`guru_sync` correctly uses `guru -modified` when the current buffer is modified, but `guru_async` did not. I'm not sure the behavior of guru is even defined when `-modified` is provided, but nothing is written to its stdin.